### ISSUE TITLE
Customize installer for Mac

### ISSUE THREAD

**Issue by aishahsofea** [electron-userland/electron-builder#8064]
I am using electron-builder to package my electron app for Mac. I would like to allow my user to choose whether they can run the launcher app after installation is successful. Probably something like a checkbox. If it is being checked, and user click Close, the launcher will run. Is it possible at all to customize this? I can't find it anywhere in the documentation.

**Comment by aishahsofea** [comment]
@mmaietta yes, it is a `.pkg` target, and it does not require drag-dropping. I will need to go through these installation steps and clicking Close at the end will just close the installer. I'd like to add a checkbox at this step. Another alternative could be to automatically launch the launcher when closing the installer. 

<img width="624" alt="Screenshot 2024-02-21 at 12 13 06 AM" src="https://github.com/electron-userland/electron-builder/assets/37907743/cf90ff99-e85e-4168-bd72-08c19deb9af6">

**Comment by mmaietta** [comment]
Can you try writing a `postinstall` file to be included with the `.pkg`?
I found this resource on a simple implementation
https://stackoverflow.com/a/42850950

It doesn't seem possible to change the installer dialog AFAICT. (Happy to be proven wrong though 🙂 )

**Comment by KaminoRyo** [comment]
@aishahsofea 
Oh, The approach suggested by mmaietta should be fine, but there is another problem. Also see #8063

**Comment by aishahsofea** [comment]
@mmaietta

> Can you try writing a `postinstall` file to be included with the `.pkg`? I found this resource on a simple implementation https://stackoverflow.com/a/42850950
> 
> It doesn't seem possible to change the installer dialog AFAICT. (Happy to be proven wrong though 🙂 )

That works, I could automatically run the launcher post installation. But i still want to give user the choice whether to run it or not. I'm thinking that maybe it's possible to create a script inside `postinstall` file, where it will open a small popup window, and handles the UI & logic in this new window. @KaminoRyo do you know if this is possible?

**Comment by KaminoRyo** [comment]
Sorry, but I don't know about that. After searching, I found the following articles to be helpful.

- [How to show GUI dialog window from shell script on a Mac by Artem Sapegin](https://sapegin.me/blog/show-gui-dialog-from-shell/)
- [macos - How do I make a Mac Terminal pop-up/alert? Applescript? - Stack Overflow](https://stackoverflow.com/questions/5588064/how-do-i-make-a-mac-terminal-pop-up-alert-applescript)

**Comment by mmaietta** [comment]
Well, pkg scripts only allow a pre/postinstall hook, so it makes sense that it's running as part of the "install" phase. `pkg` installers are purely handled by macos system, I don't think it's possible to have it run after hitting the close confirmation on the installer dialog
